src/ucp/core/ucp_mm.c

```diff
@@ -557,18 +582,18 @@ ucp_memh_init_uct_reg(ucp_context_h context, ucp_mem_h memh, unsigned uct_flags)
         if (status != UCS_OK) {
             goto err;
         }
+
+        memh->reg_id = context->next_memh_reg_id++;
```

Comment:
why need here?

Reply:
what if we register just some new UCT MDs below (when rcache is present)?
don't we need to increase the counter as well?